packages/fast-usdc/proposals/start-fast-usdc.js

```diff
@@ -0,0 +1,144 @@
+// TODO move file to a better parent https://github.com/Agoric/agoric-sdk/issues/9473
+import { AmountMath } from '@agoric/ertp/src/amountMath.js';
+import { deeplyFulfilledObject, makeTracer } from '@agoric/internal';
+import { E } from '@endo/far';
+
+const trace = makeTracer('StartFastUsdc', true);
+const { Fail } = assert;
+
+/**
+ * @import {Installation, Instance} from '@agoric/zoe/src/zoeService/utils.js';
+ * @import {Board} from '@agoric/vats';
+ * @import {FastUsdcSF} from '../src/fast-usdc.contract.js';
+ * @import {ManifestBundleRef} from '@agoric/deploy-script-support/src/externalTypes.js';
+ * @import {BootstrapManifest} from '@agoric/vats/src/core/lib-boot.js';
+ */
+
+/**
+ * @param {string} path
+ * @param {{
+ *   chainStorage: ERef<StorageNode | null>;
+ *   board: ERef<Board>;
+ * }} io
+ */
+const makePublishingStorageKit = async (path, { chainStorage, board }) => {
+  const root = await chainStorage;
+  root || Fail`chainStorage null case is vestigial`;
+  const storageNode = await E(chainStorage)?.makeChildNode(path);
+
+  const marshaller = await E(board).getPublishingMarshaller();
+  return { storageNode, marshaller };
+};
+
+/**
+ * @param {BootstrapPowers & {
+ *   installation: PromiseSpaceOf<{
+ *     fastUsdc: Installation<FastUsdcSF>;
+ *   }>;
+ *   instance: PromiseSpaceOf<{
+ *     fastUsdc: Instance<FastUsdcSF>;
+ *   }>;
+ *   brand: PromiseSpaceOf<{ USDC: Brand<'nat'> }>;
+ *   issuer: PromiseSpaceOf<{ USDC: Issuer<'nat'> }>;
+ * }} powers
+ */
+export const startFastUsdc = async ({
+  consume: {
+    agoricNames,
+    board,
+    chainStorage,
+    chainTimerService: timerService,
+    localchain,
+    cosmosInterchainService,
+    startUpgradable,
+  },
+  installation: {
+    consume: { fastUsdc },
+  },
+  instance: {
+    produce: { fastUsdc: produceInstance },
+  },
+}) => {
+  trace('startFastUsdc');
+  await null;
+  const USDC = {
+    brand: await E(agoricNames).lookup('brand', 'USDC'),
+    issuer: await E(agoricNames).lookup('issuer', 'USDC'),
+  };
+  const terms = harden({
+    poolFee: AmountMath.make(USDC.brand, 100n), // TODO: parameterize
+    contractFee: AmountMath.make(USDC.brand, 30n),
+  });
+  const { storageNode, marshaller } = await makePublishingStorageKit(
+    'fastUsdc',
+    { board, chainStorage },
+  );
+  assert(storageNode);
+
+  const privateArgs = await deeplyFulfilledObject(
+    harden({
+      agoricNames,
+      localchain,
+      orchestrationService: cosmosInterchainService,
+      storageNode,
+      timerService,
+      marshaller,
+    }),
+  );
+
+  const startOpts = {
+    label: 'fastUsdc',
+    installation: fastUsdc,
+    issuerKeywordRecord: harden({ USDC: USDC.issuer }),
+    terms,
+    privateArgs,
+  };
+
+  const { instance } = await E(startUpgradable)(startOpts);
+
+  produceInstance.reset();
+  produceInstance.resolve(instance);
+  trace('startFastUsdc done', instance);
+};
+harden(startFastUsdc);
```

Comment:
nit: maybe create `PoolShares` here?

Reply:
Good point. The fact that CI is green without doing that highlights that this module isn't tested at all. I'll leave it out until we get to it, in milestone 2.